### PR TITLE
UX: Preview multiple color schemes in wizard.

### PR DESCRIPTION
**Pull request by misaka4e21**
It was a dropdown to provide choices of color schemes,
and only one scheme could be shown.
With this commit, multiple color scheme previews can be displayed on
one page at the same time, making admins choose color schemes more
easily.

Theme preview windows are shrinked.

Imported default color schemes.

### CODE REVIEW

**Comment by misaka4e21** [comment]
default
default

**Comment by coding-horror** [comment]
This looks good! One very very tiny teeny small tweak, change "welcome to Discourse" to "Discussion topic" as we now have two sets of text that say "Discourse", it is a bit much. ;)

**Comment by misaka4e21** [comment]
default
default

**Comment by coding-horror** [comment]
Very nice 😍

File: app/assets/javascripts/wizard/components/theme-preview.js.es6 [hunk 1]
```diff
@@ -46,41 +46,44 @@ export default createPreviewComponent(659, 320, {
...
     }
 
+    // Share Button
+    ctx.beginPath();
+    ctx.rect(margin, line + lineHeight, width * 0.14, height * 0.14);
+    ctx.fillStyle = darkLightDiff(colors.primary, colors.secondary, 90, 65);
+    ctx.fill();
+    ctx.fillStyle = chooseDarker(colors.primary, colors.secondary);
+    ctx.font = `${bodyFontSize}em 'Arial'`;
+    ctx.fillText("Share", margin + width/55, line + lineHeight * 1.85);
```
Comment (jjaffeux) [inline app/assets/javascripts/wizard/components/theme-preview.js.es6]: We use prettier now, and this line should be formatted like this:

```
ctx.fillText("Share", margin + width / 55, line + lineHeight * 1.85);
```

I recommend you setup some prettier integration in your editor which will format on save, there are various prettier offenses in this PR. Let me know if you need help.

https://prettier.io/

File: app/assets/javascripts/wizard/components/theme-preview.js.es6 [hunk 2]
```diff
@@ -46,41 +46,44 @@ export default createPreviewComponent(659, 320, {
     ctx.beginPath();
     ctx.fillStyle = colors.primary;
     ctx.font = `bold ${titleFontSize}em 'Arial'`;
-    ctx.fillText("Welcome to Discourse", margin, height * 0.25);
+    ctx.fillText("Discussion topic", margin, height * 0.3);
```
Comment (jjaffeux) [inline app/assets/javascripts/wizard/components/theme-preview.js.es6]: Should we use i18n for all the texts?

File: app/assets/javascripts/wizard/templates/components/theme-previews.hbs [hunk 3]
```diff
@@ -0,0 +1,11 @@
+<ul class="grid">
+  {{#each field.choices as |choice|}}
+    <li>
+      {{theme-preview colors_id=choice.id wizard=wizard}}
```
Comment (jjaffeux) [inline app/assets/javascripts/wizard/templates/components/theme-previews.hbs]: where is `colors_id` used in the `theme-preview` please?

Comment (jjaffeux) [inline app/assets/javascripts/wizard/templates/components/theme-previews.hbs]: should it be `colorsId` ?

File: app/assets/stylesheets/wizard.scss [hunk 4]
```diff
@@ -106,6 +111,29 @@ body.wizard {
   }
 }
 
+.wizard-step-colors {
+  ul.grid {
```
Comment (jjaffeux) [inline app/assets/stylesheets/wizard.scss]: why do we have `ul` selector here, and no just the `.grid` ?

**Comment by jjaffeux** [comment]
Hi,

this is looking great! I made a few comments on code.

And have a few points on the feature itself:
- couldn’t we make it so we can click on the component itself (with a cursor: pointer) instead of having to click on a radio?
- it seems a bit not horizontally aligned
- there's also more space between items vertically than horizontally (if we remove the name and the radio button it will look nicer I think)
- if we make it clickable, maybe add some box-shadow on hover

If you don’t have time for these changes, the PR is totally mergeable without changing the UI :+1: great work

**Comment by jjaffeux** [comment]
When I choose a theme it seems to be applied, but doesn’t seem to work:

screen shot 2018-07-20 at 02 11 17

**Comment by SamSaffron** [comment]
@misaka4e21 did you update to use @awesomerobot's selected color schemes?

File: app/models/color_scheme.rb [hunk 5]
```diff
@@ -16,6 +16,78 @@ class ColorScheme < ActiveRecord::Base
       "danger" =>            'e45735',
       "success" =>           '1ca551',
       "love" =>              'fa6c8d'
+    },
```
Comment (SamSaffron) [inline app/models/color_scheme.rb]: lets add a small comment next to each color scheme thanking the particular author of the color scheme for the contribution.

**Comment by SamSaffron** [comment]
ok we are almost good to go once @misaka4e21 gets the comment in @jjaffeux you are good to merge

File: app/assets/stylesheets/wizard.scss [hunk 6]
```diff
@@ -40,7 +40,9 @@ body.wizard {
 }
 
 .wizard-warning {
-  font-family: sans-serif, p {
+  font-family: sans-serif,
```
Comment (jjaffeux) [inline app/assets/stylesheets/wizard.scss]: should be a ;

**Comment by jjaffeux** [comment]
There was multiple styling and prettier offenses, I corrected them from my branch and pushed it here: https://github.com/discourse/discourse/commit/7a3c5410770ec0752f72ce13f87bef6601c4aa0d

Great work @misaka4e21 thanks.

**Pull request closed without merging**